sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCScan.scala

```diff
@@ -53,6 +53,7 @@ case class JDBCScan(
         relation.buildScan(columnList, prunedSchema, pushedPredicates, groupByColumns, tableSample,
           pushedLimit, sortOrders, pushedOffset)
       }
+      override def name: String = "JDBC Relation"
```

Comment:
can't we just override `toString` here? Why do we need a new trait?

Reply:
Adding new traits adds complexity to the system and we need to justify it.

Reply:
It is a little bit complicate, because, node name visible in Spark UI uses toString method for relation. On the other side, for QPL, if we use toString instead of class name, entire message (which can be huge) will be serialized to name property. I can give you references of our fork in private chat, I was considering option to add this trait just to our fork, but my concern is that some contributor will add some similar trait to Spark and then the situation will be even worse in our fork.

Reply:
How about the idea from @MaxGekk ? https://github.com/apache/spark/pull/45259/files#r1504323261

We can add e.g. `JDBCFallbackRelation` instead of using anonymous class.

Reply:
Yes, that is good option. I was wondering what to implement, on the end I come with this because of Snowflake and Redshift for federation, is ok to implement fallback relation for all 3 datasource types (Redshift, Snowflake, JDBC)?

Reply:
I think anonymous class should only be used for simple cases. It's always better to have a normal class with descriptive names.

Reply:
Thanks, I will soon update it